interactors/advanced-search.js

```diff
@@ -3,17 +3,23 @@ import Select from './select';
 import TextArea from './textarea';
 import HTML from './baseHTML';
 
+const getRowIndex = el => parseInt(el.getAttribute('data-row-index'), 10);
+
 export const AdvancedSearchRow = HTML.extend('advanced search row')
   .selector('[class*=AdvancedSearchRow-]')
   .filters({
-    index: (el) => parseInt(el.getAttribute('data-row-index'), 10),
-    query: (el) => el.querySelector('textarea').getAttribute('value'),
-    bool: (el) => el.querySelectorAll('select')[0].value,
-    option: (el) => el.querySelectorAll('select')[1].value,
+    index: el => getRowIndex(el),
+    query: el => el.querySelector('textarea').getAttribute('value'),
+    bool: el => el.querySelectorAll(`select#advanced-search-bool-${getRowIndex(el)}`).value,
+    matchOption: el => el.querySelectorAll(`select#advanced-search-match-${getRowIndex(el)}`).value,
+    option: el => el.querySelectorAll(`select#advanced-search-option-${getRowIndex(el)}`).value,
   })
   .actions({
     fillQuery: ({ find }, value) => find(TextArea({ ariaLabel: 'Search for' })).fillIn(value),
-    selectBoolean: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-bool-${rowIndex}` })).choose(value),
+    selectBoolean: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-bool-${rowIndex}` }))
+      .choose(value),
+    selectMatchOption: ({ find }, rowIndex, value) => find(Select({ id: `advanced-search-match-${rowIndex}` }))
```

Comment:
Here, the actual id of match option select would be `advanced-match-${rowIndex}`